### DISCUSSION TITLE
Self-hosting Neon with Azure Blob Storage Integration

### DISCUSSION THREAD

**Discussion by muandane** [neondatabase/neon#5192] [Q&A]
Hi everyone,

Our organization is eager to self-host Neon for internal testing and, potentially production. Initially, our plan was to utilize an Azure VM instance, though we're contemplating switching to AKS for scalability. However, a key concern is the cost associated with using S3 instances for data storage. This expense becomes significant, especially as we are required to use Minio as an intermediary for S3 storage.

Hence, my question is: Is there a possibility of integrating support for Azure Blob storage within Neon? This would offer us a more cost-effective solution, as direct Azure Blob storage usage could potentially reduce expenses compared to using S3 storage via Minio.

Your insights and feedback on this matter would be greatly appreciated Thank you.

**Comment by harishoss** [comment]
unrelated to your main question. Why is using Minio as an intermediary more expensive than directly using Azure blob storage?

**Reply by muandane** [reply]
So we'd have to use minio in aks with our own pvc it'll be stored into an azure disk which is still part of the azure storage ecosystem so we'll be paying for azure disk storage + minio licence.
So if you use neon in AKS you could use directly blob storage for storing Postgres data.

**Comment by 9to1url** [comment]
I am also interesting on similar use cases, can I use Ceph instead of S3?
